CRM/Financial/BAO/FinancialItem.php

```diff
@@ -109,7 +109,10 @@ public static function add($lineItem, $contribution, $taxTrxnID = FALSE, $trxnId
     }
     else {
       $accountRelName = 'Income Account is';
-      if (property_exists($contribution, 'revenue_recognition_date') && !CRM_Utils_System::isNull($contribution->revenue_recognition_date)) {
+      if (($itemStatus == array_search('Unpaid', $financialItemStatus)) &&
```

Comment:
Just raising a uninformed question in case it hasn't been tested: when reversing out deferred revenue items, is this code used, and would it be appropriate to provide the revenue recognition date and deferred revenue account?